R/display.R

```diff
@@ -1,74 +1,136 @@
+namedlist <- function() setNames(list(), character(0))
+
 base_display <- function(data, metadata) {
     warning("IR_display can only be used from the IPython R kernel and R magic.")
 }
 
 #'Display data by mimetype, with optional alternative representations.
 #'
-#' @param data A named list mapping mimetypes to content (base64 encoded for binary data)
-#' @param metadata A named list mapping mimetypes to named lists of specific metadata
+#' @param data  A named list mapping mimetypes to content (base64 encoded for binary data)
+#' @param metadata  A named list mapping mimetypes to named lists of specific metadata
 #' @export
-display <- function(data, metadata=NULL) {
+display <- function(data, metadata = NULL) {
     base_display(data, metadata)
 }
 
 #'Display a single type of data
 #'
-#' @param mimetype The mimetype of the data
-#' @param  content The data as a length 1 string vector
-#' @param  metadata A named list of metadata
+#' @param mimetype  The mimetype of the data
+#' @param content  The data as a length 1 string vector
+#' @param metadata  A named list of metadata
+#' @export
+display1 <- function(mimetype, content, metadata = NULL) {
+    data <- list()
+    data[[mimetype]] <- content
+    base_display(data, metadata)
+}
+
+#'Display a set of formatted objects that the Jupyter frontend/output can select from
+#'
+#' @param metadata  Initial metadata. Mimetype-wise data will be added
+#' @param ...  Mimetypes and content.
+#' 
+#' You can either give raw vectors or character vectors as content.
+#' 
+#' Also you can give a list of which the first element will be content, the rest metadata.
+#' 
+#' @examples
+#' \dontrun{
+#' display_alternatives(
+#'   'text/html'       = '<em>Wop</em>'
+#'   'application/pdf' = base64encode('test.pdf'),
+#'   'image/png'       = list(some.png.data, width = 800, height = 600),
+#'   'image/svg+xml'   = '<svg></svg>')
+#' }
+#' 
+#' @importFrom base64enc base64encode
 #' @export
-display1 <- function(mimetype, content, metadata=NULL) {
-    data = list()
-    data[mimetype] = content
+display_alternatives <- function(..., metadata = namedlist()) {
```

Comment:
i’m going all python zen here: expicit is better than implicit, yet practicality beats purity.

but `display_alternatives` is more related to `display1`, because it knows about mimetypes. `display` is simply “do everything manually”, `display1` and `display_alternatives` are “display things using mimetypes”, and `display_*` are convenience functions to save people from remembering all those mimetypes.